src/client/dfuse/dfuse_core.c

```diff
@@ -517,6 +670,35 @@ dfuse_cont_open(struct dfuse_projection_info *fs_handle, struct dfuse_pool *dfp,
 						strerror(rc));
 				D_GOTO(err_close, rc);
 			}
+
+			if (fs_handle->dpi_info->di_caching) {
+				rc = dfuse_cont_get_cache(dfc);
+				if (rc == ENODATA) {
+					/* If there is no container specific
+					 * attributes then use defaults
+					 */
+					DFUSE_TRA_INFO(dfc,
+						       "Using default caching values");
+					dfc->dfc_attr_timeout = 1;
+					dfc->dfc_dentry_timeout = 1;
+					dfc->dfc_dentry_dir_timeout = 5;
+					dfc->dfc_ndentry_timeout = 5;
+					dfc->dfc_data_caching = true;
+					rc = 0;
+				} else if (rc != 0) {
+					D_GOTO(err_close, rc);
+				}
+			} else {
+				DFUSE_TRA_INFO(dfc,
+					"Caching disabled");
+			}
+		} else {
+			/* Set attributes for containers created via mkdir */
+			dfc->dfc_attr_timeout = 1;
+			dfc->dfc_dentry_timeout = 1;
+			dfc->dfc_dentry_dir_timeout = 5;
+			dfc->dfc_ndentry_timeout = 5;
```

Comment:
As L685.  I'll merge these two code-blocks into a "dfuse_set_cache_defaults()" function on repush to make this clear.

Reply:
Done.